codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -575,11 +575,8 @@ private State readHeaders(ByteBuf buffer) {
                 char firstChar = line.charAt(0);
                 if (name != null && (firstChar == ' ' || firstChar == '\t')) {
                     String trimmedLine = line.toString().trim();
-                    StringBuilder buf = new StringBuilder(value.length() + trimmedLine.length() + 1);
-                    buf.append(value)
-                       .append(' ')
-                       .append(trimmedLine);
-                    value = buf.toString();
+                    String valueStr = String.valueOf(value);
+                    value = valueStr + ' ' + trimmedLine;
```

Comment:
@normanmaurer this will break `OptimizeStringContact` optimization. As part of this PR I moved any operations (even it is `String.valueOf(value)`) out from concat operation. Here is benchmark that shows that:

```
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.NANOSECONDS)
@Fork(1)
@State(Scope.Thread)
@Warmup(iterations = 5, time = 1)
@Measurement(iterations = 5, time = 1)
public class BuilderVsRegular3Concat {

    @Param({"Some String"})
    private String s;
    @Param({"Some Another Char sequence2"})
    String line;

    @Benchmark
    public String builderWithLength() {
        return new StringBuilder(s.length() + 1 + line.length())
                .append(String.valueOf(s))
                .append(' ')
                .append(line)
                .toString();
    }

    @Benchmark
    public String builder() {
        return new StringBuilder()
                .append(String.valueOf(s))
                .append(' ')
                .append(line)
                .toString();
    }

    @Benchmark
    public String naiveWithLocalVariable() {
        String loc = String.valueOf(s);
        return loc + ' ' + line;
    }

}
```

```
Benchmark                                                            (line)          (s)  Mode  Cnt   Score   Error  Units
BuilderVsRegular3Concat.builder                 Some Another Char sequence2  Some String  avgt    5  54.910 ± 9.031  ns/op
BuilderVsRegular3Concat.builderWithLength       Some Another Char sequence2  Some String  avgt    5  45.883 ± 5.666  ns/op
BuilderVsRegular3Concat.naiveWithLocalVariable  Some Another Char sequence2  Some String  avgt    5  21.914 ± 4.014  ns/op
```

This is valid for all JVM versions.